### PR TITLE
Add updates for CUDA 12 libraries

### PR DESCRIPTION
**Pull request by mlxd**
### Before submitting

Please complete the following checklist when submitting a PR:

- [x] All new features must include a unit test.
      If you've fixed a bug or added code that should be tested, add a test to the
      [`tests`](../tests) directory!

- [x] All new functions and code must be clearly commented and documented.
      If you do make documentation changes, make sure that the docs build and
      render correctly by running `make docs`.

- [x] Ensure that the test suite passes, by running `make test`.

- [x] Add a new entry to the `.github/CHANGELOG.md` file, summarizing the
      change, and including a link back to the PR.

- [x] Ensure that code is properly formatted by running `make format`. 

When all the above are checked, delete everything above the dashed
line and fill in the pull request template.

------------------------------------------------------------------------------------------------------------

**Context:** This PR updates the reported error messages from CUDA specific libraries, and avoids unversioned libs from being included in wheel builds (such as when using the NVHPC toolkit).

**Description of the Change:** As above. Ensure better error messaging and reporting on complex systems (such as Perlmutter).

**Benefits:**

**Possible Drawbacks:**

**Related GitHub Issues:**

### CODE REVIEW

**Review by jay-selby — APPROVED** [review]
LGTM

File: pennylane_lightning/core/src/simulators/lightning_gpu/utils/cuError.hpp [hunk 1]
```diff
@@ -85,25 +85,28 @@ static const std::string GetCuBlasErrorString(const cublasStatus_t &err) {
...
     case CUBLAS_STATUS_ARCH_MISMATCH:
-        result = "CUDA device architecture mismatch";
+        result = "cuBLAS CUDA device architecture mismatch";
         break;
     case CUBLAS_STATUS_MAPPING_ERROR:
         result = "cuBLAS mapping error";
         break;
     case CUBLAS_STATUS_INTERNAL_ERROR:
-        result = "Internal cuBLAS error";
+        result = "cublas internal error";
```
Comment (vincentmr) [inline pennylane_lightning/core/src/simulators/lightning_gpu/utils/cuError.hpp:97]:
```suggestion
        result = "cuBLAS internal error";
```

File: pennylane_lightning/core/src/simulators/lightning_gpu/utils/cuError.hpp [hunk 2]
```diff
@@ -85,25 +85,28 @@ static const std::string GetCuBlasErrorString(const cublasStatus_t &err) {
...
         break;
     case CUBLAS_STATUS_NOT_SUPPORTED:
-        result = "Unsupported operation/device";
+        result = "cuBlas Unsupported operation/device";
```
Comment (vincentmr) [inline pennylane_lightning/core/src/simulators/lightning_gpu/utils/cuError.hpp:100]:
```suggestion
        result = "cuBLAS Unsupported operation/device";
```

File: pennylane_lightning/core/src/simulators/lightning_gpu/utils/cuError.hpp [hunk 3]
```diff
@@ -85,25 +85,28 @@ static const std::string GetCuBlasErrorString(const cublasStatus_t &err) {
...
         break;
     case CUBLAS_STATUS_EXECUTION_FAILED:
-        result = "GPU program failed to execute";
+        result = "cuBlas GPU program failed to execute";
```
Comment (vincentmr) [inline pennylane_lightning/core/src/simulators/lightning_gpu/utils/cuError.hpp:103]:
```suggestion
        result = "cuBLAS GPU program failed to execute";
```

File: pennylane_lightning/core/src/simulators/lightning_gpu/utils/cuError.hpp [hunk 4]
```diff
@@ -85,25 +85,28 @@ static const std::string GetCuBlasErrorString(const cublasStatus_t &err) {
...
+        break;
+    case CUBLAS_STATUS_LICENSE_ERROR:
+        result = "cuBlas license error";
```
Comment (vincentmr) [inline pennylane_lightning/core/src/simulators/lightning_gpu/utils/cuError.hpp:106]:
```suggestion
        result = "cuBLAS license error";
```

File: pennylane_lightning/core/src/simulators/lightning_gpu/utils/cuError.hpp [hunk 5]
```diff
@@ -85,25 +85,28 @@ static const std::string GetCuBlasErrorString(const cublasStatus_t &err) {
...
         break;
     default:
-        result = "Status not found";
+        result = "cuBlas status not found. Error code=" + std::to_string(err);
```
Comment (vincentmr) [inline pennylane_lightning/core/src/simulators/lightning_gpu/utils/cuError.hpp:109]:
```suggestion
        result = "cuBLAS status not found. Error code=" + std::to_string(err);
```

File: pennylane_lightning/core/src/simulators/lightning_gpu/utils/cuError.hpp [hunk 6]
```diff
@@ -142,7 +145,7 @@ static const std::string GetCuSparseErrorString(const cusparseStatus_t &err) {
         result = "The resources are not sufficient to complete the operation.";
         break;
     default:
-        result = "Status not found";
+        result = "cuSparse status not found. Error code=" + std::to_string(err);
```
Comment (vincentmr) [inline pennylane_lightning/core/src/simulators/lightning_gpu/utils/cuError.hpp:148]:
```suggestion
        result = "cuSPARSE status not found. Error code=" + std::to_string(err);
```

**Review by vincentmr — APPROVED** [review]
Looks good to me. I'd just suggest uniformizing the case for cuBLAS, cuSPARSE and cuStateVec error messages. Either all lowercase, or trademark case maybe?

**Pull request merged**